dubbo-rpc/dubbo-rpc-api/src/main/java/org/apache/dubbo/rpc/filter/GenericFilter.java

```diff
@@ -191,6 +192,26 @@ public Result invoke(Invoker<?> invoker, Invocation inv) throws RpcException {
                                 args.length + " and type is" +
                                 args[0].getClass().getName());
                     }
+                }  else if (ProtocolUtils.isNativeStreamGenericSerialization(generic)) {
+                    for (int i = 0; i < args.length; i++) {
+                        if (byte[].class == args[i].getClass()) {
+                            try (UnsafeByteArrayInputStream is = new UnsafeByteArrayInputStream((byte[]) args[i])) {
+                                args[i] = applicationModel.getExtensionLoader(Serialization.class)
+                                    .getExtension(invoker.getUrl().getParameter(Constants.SERIALIZATION_KEY))
+                                    .deserialize(null, is).readObject(method.getParameterTypes()[i]);
+                            } catch (Exception e) {
+                                throw new RpcException("Deserialize argument [" + (i + 1) + "] failed.", e);
+                            }
+                        } else {
+                            throw new RpcException(
+                                "Generic serialization [" +
+                                    invoker.getUrl().getParameter(Constants.SERIALIZATION_KEY) +
+                                    "] only support message type " +
+                                    byte[].class +
+                                    " and your message type is " +
+                                    args[i].getClass());
+                        }
+                    }
```

Comment:
It seems like using byte array as a serialization input. In my opinion, it would be better to being customized in a custom `Filter` not in Dubbo framework.

Reply:
ensure data types in runtime by an external filter is more compatible

Reply:
@AlbumenJ @kimmking Thanks to both of you for your opinions, we will try it. I will close this PR first.